server.go

```diff
@@ -166,8 +175,47 @@ func NewFromConfig(conf Config) (ret Server, err error) {
 	ret.HTTPAddr = conf.HTTPAddress
 	ret.ForwardAddr = conf.ForwardAddress
 
+	ret.shutdownChan = make(chan struct{})
```

Comment:
As with my other common about config, we want to allow the user the option of not running this by default. Can we skip this bit if there's no TcpAddress?